### ISSUE TITLE
[Bug]: [restful]Create collection can success with dbName which has not been created, but list collections with same dbName failed

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#25509]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20230710-4dd02836
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

create collection
```
POST http://10.100.36.162:19530/v1/vector/collections/create
200
40 ms
POST /v1/vector/collections/create HTTP/1.1
Content-Type: application/json
Authorization: Bearer root:Milvus
User-Agent: PostmanRuntime/7.32.3
Accept: */*
Postman-Token: d018aff4-c89b-47f4-be5b-1a6d7bc01558
Host: 10.100.36.162:19530
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Content-Length: 132
 
{
"collectionName": "book",
"dbName": "demo",
"dimension": 2,
"metricsType": "L2",
"vectorField": "book_intro"
}
 
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
Date: Wed, 12 Jul 2023 02:36:18 GMT
Content-Length: 22
 
{"code":200,"data":{}}
```

list collection
```
GET http://10.100.36.162:19530/v1/vector/collections?dbName=demo
200
18 ms
GET /v1/vector/collections?dbName=demo HTTP/1.1
Authorization: Basic cm9vdDpNaWx2dXM=
User-Agent: PostmanRuntime/7.32.3
Accept: */*
Postman-Token: 0d020f99-4de3-4ddf-b5ae-c1b07549afd2
Host: 10.100.36.162:19530
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
 
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
Www-Authenticate: Basic realm="restricted", charset="UTF-8"
Date: Wed, 12 Jul 2023 02:37:54 GMT
Content-Length: 104
 
{"code":400,"error":"ShowCollections failed: database:demo not found","message":"show collections fail"}
```

### Expected Behavior

the requests all successful or all fail

### Anything else?

Since Restful does not have an API to manage the database, we can not use the database if we choose Restful as the only tech stack to develop with Milvus.

We need to use SDK like pymilvus to create a database and then go back to using restful API with this database.

In short, using restful API can not get a complete lifecycle, and need to be used in conjunction with other SDKs.

**Comment by zhuwenxing** [comment]
I used pymilvus to list the db and found that the `demo` db was not created.

**Comment by zhuwenxing** [comment]
The parameter "dbName" does not take effect, even if it is modified, it still uses the default value.

So the database is not supported yet.
